doc/releases/changelog-dev.md

```diff
@@ -73,6 +73,9 @@
   method for a provided device and interface, in human-readable format.
   [(#2533)](https://github.com/PennyLaneAI/pennylane/pull/2533)
 
+* Sparse Hamiltonians representation has changed from COOrdinate (COO) to Compressed Sparse Row (CSR) format.
+[(#2561)](https://github.com/PennyLaneAI/pennylane/pull/2561)
```

Comment:
@AmintorDusko could you briefly expand on why this was done, and the impact :)

Reply:
Sure! Thank you for your suggestion, @josh146.